src/telliot/reporter/interval.py

```diff
@@ -23,76 +25,118 @@ def __init__(
-
-            datapoint = datafeed.source.latest
-            v, t = datapoint
-
-            if v is not None:
-                query = datafeed.query
-
-                if query:
-                    encoded_value = query.value_type.encode(v)
-                    request_id_str = "0x" + query.query_id.hex()
-                    extra_gas_price = 0
-
-                    for _ in range(retries + 1):
-                        (
-                            status,
-                            transaction_receipt,
-                            gas_price,
-                        ) = self.submitter.submit_data(
-                            encoded_value, request_id_str, extra_gas_price
-                        )
-
-                        if transaction_receipt and status.ok:
-                            transaction_receipts.append(transaction_receipt)
-                            break
-                        elif (
-                            not status.ok
-                            and status.error
-                            and "replacement transaction underpriced" in status.error
-                        ):
-                            extra_gas_price += gas_price
-                        else:
-                            extra_gas_price = 0
+        status = ResponseStatus()
+        gas_price_gwei = await fetch_gas_price()
 
-                else:
-                    print(
-                        f"Skipping submission for {repr(datafeed)}, "
-                        f"no query for datafeed."
-                    )  # TODO logging
+        transaction_receipts: List[
+            Tuple[Optional[AttributeDict[Any, Any]], ResponseStatus]
+        ] = []
+
+        user = self.endpoint.web3.eth.account.from_key(self.private_key).address
+        is_staked, read_status = await self.master.read("getStakerInfo", _staker=user)
+
+        if (not read_status.ok) or (is_staked is None):
+            status.error = "unable to read reporter staker status: " + read_status.error  # type: ignore # error won't be none # noqa: E501
+            status.e = read_status.e
+            transaction_receipts.append((None, status))
+
+        else:
+            print(is_staked[0])
+
+            # Status 1: staked
+            if is_staked[0] == 1:
+                jobs = []
+                for datafeed in self.datafeeds:
+                    job = asyncio.create_task(datafeed.source.fetch_new_datapoint())
+                    jobs.append(job)
+
+                _ = await asyncio.gather(*jobs)
+
+                for datafeed in self.datafeeds:
+
+                    datapoint = datafeed.source.latest
+                    v, t = datapoint
+
+                    if v is not None:
+                        query = datafeed.query
+
+                        if query:
+                            value = query.value_type.encode(v)
+                            query_id = query.query_id
+                            query_data = query.query_data

```

Comment:
Could remove these two var assignments and just pass those attributes in the function call on line 98 & 101.

Reply:
True, but I think it's more readable in the current format. @clamdad thoughts?